fuse/data/ops/ops_aug_common.py

```diff
@@ -179,3 +180,70 @@ def __init__(self, op: OpBase, kwargs_per_step_to_add: Sequence[dict]):
         :param kwargs_per_step_to_add: sequence of arguments (kwargs format) specific for a single repetition. those arguments will be added/overide the kwargs provided in __call__() function.
         """
         super().__init__(OpSample(op), kwargs_per_step_to_add)
+
+
+class OpRandCrop(OpBase):
+    """
+    Crops a given string based on a randomly generated ratio between 0 and crop_max_ratio.
+
+    Can be used as an augmentation method for cropping a protein/any other input
+    """
+
+    def __init__(
+        self,
+        crop_max_ratio: float = 0.3,
+        crop_both_sides: bool = True,
+        crop_left_only: bool = False,
+        crop_right_only: bool = False,
+    ):
+        """
+        :param crop_max_ratio: The maximum ratio of the string length that can be cropped in total.
+                A random ratio between 0 and crop_max_ratio will be generated for cropping.
+
+        :param crop_both_sides : If True, crop the string from both sides.
+
+        :param crop_left_only : If True, crop the string from the left side only.
+
+        :param crop_right_only : If True, crop the string from the right side only.
+
+        """
+        super().__init__()
+
+        assert 0 <= crop_max_ratio <= 1.0
+        assert any([crop_both_sides, crop_left_only, crop_right_only])
+
+        self.crop_both_sides = crop_both_sides
+        self.crop_left_only = crop_left_only
+        self.crop_right_only = crop_right_only
+        self.crop_max_ratio = crop_max_ratio
+
+    def __call__(
```

Comment:
what is expected to be stored in key_in.
I assume either tensor or numpy array? mention it.
What are the dims? I assume that the only limitation is that dim 0 is the sequence dim, right?

Reply:
the sample_dict[string_key_in] is a string (the sequence), why do you assume it is a tensor or np array?

Reply:
My bad, In my mind it was on the already tokenized sequence and the code will work also with tensors or any list like sequence..
Anyway, please add in a comment that you are expecting a sequence represented in string.